src/test/java/com/heatheryou/httpserver/SetupTest.java

```diff
@@ -0,0 +1,47 @@
+package com.heatheryou.httpserver;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.Assert.*;
+
+public class SetupTest {
+    MockSystemOutput mockOutput;
+    String[] args;
+    Setup setup;
+
+    @Before
+    public void setup() {
+        mockOutput = new MockSystemOutput();
+        args = new String[]{"random", "5000"};
+        setup = new Setup(args, new CommandLineArgs(), mockOutput);
+    }
+
+    @Test
+    public void executeReturnsPortNumberIfArgumentContainsOnlyIntegers() {
+        String[] newArgs = new String[]{"3000"};
+        setup = new Setup(newArgs, new CommandLineArgs(), mockOutput);
+
+        int actual = setup.execute();
+        int expected = 3000;
+        assertEquals(actual, expected);
+    }
+
+    @Test
+    public void executeExitsApplicationWithInvalidRequest() {
```

Comment:
`WithInvalidCommandLineArguments` or something to that effect?  We don't really have a request yet.

Reply:
Ahh, right right